storage/s3/src/main/java/io/aiven/kafka/tieredstorage/storage/s3/MetricCollector.java

```diff
@@ -76,35 +68,35 @@ private Sensor createSensor(final String name) {
     }
 
     @Override
-    public void collectMetrics(final Request<?> request, final Response<?> response) {
-        final AWSRequestMetrics awsRequestMetrics = request.getAWSRequestMetrics();
-        if (awsRequestMetrics == null) {
-            return;
-        }
-        final List<Object> requestTypes = awsRequestMetrics.getProperty(AWSRequestMetrics.Field.RequestType);
-        if (requestTypes == null || requestTypes.isEmpty()) {
-            return;
-        }
-        final Object requestType = requestTypes.get(0);
-        if (requestType == null) {
-            return;
+    public void publish(final MetricCollection metricCollection) {
+        final List<String> metricValues = metricCollection.metricValues(CoreMetric.OPERATION_NAME);
+        for (final String metricValue : metricValues) {
+            if ("GetObject".equals(metricValue)) {
+                this.getObjectRequests.record();
+            }
+            if ("UploadPart".equals(metricValue)) {
+                this.uploadPartRequests.record();
+            }
+            if ("CreateMultipartUpload".equals(metricValue)) {
+                this.createMultipartUploadRequests.record();
+            }
+            if ("CompleteMultipartUpload".equals(metricValue)) {
+                this.completeMultipartUploadRequests.record();
+            }
+            if ("PutObject".equals(metricValue)) {
+                this.putObjectRequests.record();
+            }
+            if ("DeleteObject".equals(metricValue)) {
+                this.deleteObjectRequests.record();
+            }
+            if ("AbortMultipartUpload".equals(metricValue)) {
+                this.abortMultipartUploadRequests.record();
+            }
         }
+    }
 
-        // Try to arrange them by likelihood, at least GetObjectRequest should be first.
```

Comment:
Probably makes sense to preserve this commit

Reply:
You mean comment, right?

Reply:
Yeah, sorry :)